scan-payment/src/main/java/com/getbouncer/scan/payment/analyzer/NameDetectAnalyzer.kt

```diff
@@ -0,0 +1,268 @@
+package com.getbouncer.scan.payment.analyzer
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.RectF
+import android.media.MediaScannerConnection
+import android.media.MediaScannerConnection.OnScanCompletedListener
+import android.provider.MediaStore
+import android.util.Log
+import com.getbouncer.scan.framework.Analyzer
+import com.getbouncer.scan.framework.AnalyzerFactory
+import com.getbouncer.scan.framework.ml.ssd.hardNonMaximumSuppression
+import com.getbouncer.scan.payment.ml.AlphabetDetect
+import com.getbouncer.scan.payment.ml.SSDObjectDetect
+import com.getbouncer.scan.payment.ml.SSDOcr
+import com.getbouncer.scan.payment.ml.ssd.DetectionBox
+import com.getbouncer.scan.payment.ml.toObjectDetectionCroppedImage
+import kotlinx.coroutines.CoroutineScope
+import java.util.*
+
+
+private const val NUM_PREDICTION_STRIDES = 10
+private const val NMS_THRESHOLD = 0.75F
+private const val CHAR_CONFIDENCE_THRESHOLD = 0.7
+
+
+class NameDetectAnalyzer private constructor(
+    private val ssdObjectDetect: SSDObjectDetect?,
+    private val alphabetDetect: AlphabetDetect?
+) : Analyzer<SSDOcr.SSDOcrInput, PaymentCardOcrState, NameDetectAnalyzer.Output> {
+
+    data class Output(
+        val name: String?,
+        val boxes: List<DetectionBox>?
+    )
+
+    override val name: String = "card_side_analyzer"
+
+    override suspend fun analyze(data: SSDOcr.SSDOcrInput, state: PaymentCardOcrState): Output {
```

Comment:
this can be simplified:

```suggestion
    override suspend fun analyze(
        data: SSDOcr.SSDOcrInput,
        state: PaymentCardOcrState
    ) = if (!state.runNameExtraction || ssdObjectDetect == null || alphabetDetect == null) {
        Output("", null)
    } else {
        ...
    }
```